src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleAutoRod.kt

```diff
@@ -0,0 +1,193 @@
+import net.ccbluex.liquidbounce.utils.entity.getActualHealth
+import net.ccbluex.liquidbounce.utils.inventory.Slots
+import net.minecraft.client.option.KeyBinding
+import net.minecraft.entity.Entity
+import net.minecraft.entity.LivingEntity
+import net.minecraft.item.Items
+import net.minecraft.text.Text
+
+
+object ModuleAutoRod : ClientModule("AutoRod", Category.COMBAT) {
+
+    object facingEnemy : ToggleableConfigurable(this, "FacingEnemy", true) {
+
+        object ignoreOnEnemyLowHealth : ToggleableConfigurable(this, "IgnoreOnEnemyLowHealth", true) {
+            val enemyHealthThreshold by int(
+                "EnemyHealthThreshold", 5, 1..20
+            )
+            val playerHealthThreshold by int("PlayerHealthThreshold", 5, 1..20)
+        }
+    }
+
+    init {
+        tree(facingEnemy)
+        tree(ignoreOnEnemyLowHealth)
+    }
+
+    private val enemiesNearby by int("EnemiesNearby", 1, 1..5)
+    private val targetTracker = tree(TargetTracker(PriorityEnum.DIRECTION))
+    private val escapeHealthThreshold by int("EscapeHealthThreshold", 10, 1..20)
+    private val pushDelay by int("PushDelay", 100, 50..1000)
+    private val pullbackDelay by int("PullbackDelay", 500, 50..1000)
+    private val onUsingItem by boolean("OnUsingItem", false)
+    private val pushTimer = Chronometer()
+    private val rodPullTimer = Chronometer()
+    private var rodInUse = false
+    private var switchBack: Int? = -1
+    private var range by float("Range", 5f, 1f..10f)
+
+    override fun disable() {
+        KeyBinding.setKeyPressed(mc.options.useKey.boundKey, false)
+    }
+
+    val tickHandler = tickHandler {
+        // Check if player is using rod
+        val usingRod =
+            (mc.player?.isUsingItem == true && mc.player?.mainHandStack?.item == Items.FISHING_ROD) || rodInUse
+
+        if (usingRod) {
+            // Check if rod pull timer has reached delay
+            // mc.player.fishEntity?.caughtEntity != null is always null
+
+            if (rodPullTimer.hasElapsed(pullbackDelay.toLong())) {
+                if (switchBack != -1 && mc.player?.inventory?.selectedSlot != switchBack) {
+                    // Switch back to previous item
+                    mc.player?.inventory?.selectedSlot = switchBack!!
+                    interaction.syncSelectedSlot()
+
+                } else {
+                    // Stop using rod
+                    mc.player?.stopUsingItem()
+                }
+
+                switchBack = -1
+                rodInUse = false
+
+                // Reset push timer. Push will always wait for pullback delay.
+                pushTimer.reset()
+            }
+        } else {
+            var rod = false
+
+            if (facingEnemy.enabled && mc.player?.getActualHealth()!! >= ignoreOnEnemyLowHealth.playerHealthThreshold) {
+                var facingEntity = mc.targetedEntity
+                if (facingEntity == null) {
+                    
+                    var finaltarget: Entity? = null
```

Comment:
`finaltarget` does not follow Kotlin Naming Convention as well - should be `finalTarget` instead.